### ISSUE TITLE
False Positive Warning for IDE0079 Remove Unneccessary Suppression

### ISSUE THREAD

**Issue by vsfeedback** [dotnet/roslyn#47288]
_This issue has been moved from [a ticket on Developer Community](https://developercommunity.visualstudio.com/content/problem/1167673/false-positive-warning-for-ide0079-remove-unnecces.html)._

---
[severity:I'm somewhat annoyed. A fix would be nice] [regression] [worked-in:16.7]
I use the SuppressMessageAttribute to suppress certain Resharper warnings.  Visual Studio 2019 16.8.0 preview 2 is incorrectly suggesting that these attributes can be removed because they are not used.  In fact, they are being used by Resharper.

[SuppressMessage("ReSharper", "UnusedMember.Global", Justification = "For future use.")]

Would it be possible to check the category and not step on Resharper's SuppressMessage attributes?

---
#### Feedback Bot on 8/30/2020, 08:05 PM: 

<p>We have directed your feedback to the appropriate engineering team for further evaluation. The team will review the feedback and notify you about the next steps.</p>

---
### Original Solutions
(no solutions)

**Comment by mavasani** [comment]
This is by design, and the analyzer has a configurable option to skip specific diagnostic IDs from analysis. You can specify the following in your .editorconfig file: https://github.com/dotnet/roslyn/blob/163404b1385780bee5fc8a19a9ffc8ca44fdb773/src/VisualStudio/Core/Test/Options/CSharpEditorConfigGeneratorTests.vb#L87

```ini
[*.cs]

dotnet_remove_unnecessary_suppression_exclusions = ID1, ID2
```

Basically, your options are:
1. Turn off `IDE0079` for the entire project
2. Exclude specific diagnostic IDs from `IDE0079` analysis as above.

**Comment by mavasani** [comment]
> Would it be possible to check the category and not step on Resharper's SuppressMessage attributes?

Not really - suppress message attributes are meant for suppressing _analyzer diagnostics_. Any third party component that uses the same suppression mechanisms for suppressing their own custom reported issues is relying on an unsupported suppression mechanism. Adding such a custom exclusion for a specific category used by a particular third party extension is not scalable, as there could be numerous different extensions using their own category that would then request the same.

**Comment by mavasani** [comment]
Tag @sharwell to bring this up for IDE design meeting review, I have marked this tentatively as by design.

**Comment by JoeRobich** [comment]
During this weeks design review it was decided to move forward on a proposal to implement category exclusion for these diagnostics in a manner akin to the `dotnet_remove_unnecessary_suppression_exclusions` setting.

**Comment by mavasani** [comment]
After https://github.com/dotnet/roslyn/pull/47819 is merged, you should be able to exclude entire category by setting the below editorconfig option:

```editorconfig
[*.cs]
dotnet_remove_unnecessary_suppression_exclusions = category: ReSharper
```

**Comment by drvic10k** [comment]
I think this not happening only for resharper's rules

[SuppressMessage("Microsoft.Maintainability", "CA1506", Justification = "By Design")] is marked as unneessary but when reamoved, build detects these errors

**Comment by sharwell** [comment]
@drvic10k Can you file a new issue for that? It sounds like an unrelated issue.

**Comment by drvic10k** [comment]
> @drvic10k Can you file a new issue for that? It sounds like an unrelated issue.

ok, created one here: https://github.com/dotnet/roslyn/issues/51884

**Comment by CenturySparkle** [comment]
What is the purpose of IDE0079 anyway?  I just installed VS2026 and it is flagging EVERYTHING as unnecessary.  Every SuppressMessage(), every #pragma disable, everything.  It doesn't seem to serve any purpose but to just add more work for the developer to spend time to figure out how to make it go away.

**Comment by CyrusNajmabadi** [comment]
@CenturySparkle see https://github.com/dotnet/roslyn/issues/47288#issuecomment-799399108
